nmstatectl/nmstatectl.py

```diff
@@ -55,6 +60,17 @@ def setup_subcommand_set(subparsers):
 
 def show(args):
     state = netinfo.show()
+    if args.only != '*':
+        patterns = [p for p in args.only.split(',')]
+        showinterfaces = []
+
+        for interface in state['interfaces']:
```

Comment:
Please extract it to a function, so it will look something nice like this:

for pattern in patterns:
     filtered_interfaces += _filter_interfaces(state['interfaces'], pattern)
state['interfaces'] = filtered_interfaces

Reply:
I do not see how this could work, because if `_filter_interfaces()` returns `None`, the None object would be added to the state. And the resulting function would just be
```
def _filter_interfaces(state, pattern):
    if fnmatch.fnmatch(state['name'], pattern):
        return state
```

I am no conviced that this is enough code for a funtion.

Reply:
_filter_interfaces returns a list of interfaces states, if none pass the filter, then it returns an empty list.
It is much more readable that way (IMO).